trin-core/src/portalnet/mod.rs

```diff
@@ -68,8 +113,47 @@ impl ssz::Decode for U256 {
 #[cfg(test)]
 mod test {
     use super::*;
+    use rstest::rstest;
     use ssz::{Decode, Encode};
 
+    const U256_MAX: U256 = U256([u64::MAX, u64::MAX, u64::MAX, u64::MAX]);
+    const U512_MAX: U512 = U512([u64::MAX, u64::MAX, u64::MAX, u64::MAX, 0, 0, 0, 0]);
+
+    #[rstest]
+    #[case(U256([1,2,3,4]), U512([1,2,3,4,0,0,0,0]))]
+    #[case(U256([492,4039,129,9984]), U512([492,4039,129,9984,0,0,0,0]))]
+    #[case(U256_MAX, U512_MAX)]
+    #[case(U256([u64::MAX,u64::MAX,u64::MAX-1,u64::MAX]), U512([u64::MAX,u64::MAX,u64::MAX-1,u64::MAX,0,0,0,0]))]
+    #[case(U256([0,0,0,0]), U512([0,0,0,0,0,0,0,0]))]
+    fn u256_to_u512_test(#[case] input: U256, #[case] expected: U512) {
+        let result = U512::from(input);
+        assert_eq!(expected, result);
+    }
+
+    #[rstest]
+    #[case(U512([1,2,3,4,0,0,0,0]), U256([1,2,3,4]))]
+    #[case(U512([420,69,420,69,0,0,0,0]), U256([420,69,420,69]))]
+    #[case(U512([u64::MAX,u64::MAX,u64::MAX,u64::MAX,0,0,0,0]), U256_MAX)]
+    #[case(U512([u64::MAX-1,u64::MAX,u64::MAX,u64::MAX,0,0,0,0]), U256([u64::MAX-1,u64::MAX,u64::MAX,u64::MAX]))]
+    #[case(U512([1,0,0,0,0,0,0,0]), U256([1,0,0,0]))]
+    fn u512_to_u256_test(#[case] input: U512, #[case] expected: U256) {
+        let result = input.try_into().unwrap();
+        assert_eq!(expected, result);
+    }
+
+    #[rstest]
+    #[case(U512([1,2,3,4,1,0,0,0]), U256([1,2,3,4]))]
+    #[case(U512([1,2,3,4,0,34,0,0]), U256([1,2,3,4]))]
+    #[case(U512([1,2,3,4,0,0,234,0]), U256([1,2,3,4]))]
+    #[case(U512([1,2,3,4,0,0,0,349]), U256([1,2,3,4]))]
+    #[case(U512([u64::MAX,u64::MAX,u64::MAX,u64::MAX,1,0,0,0]), U256([1,2,3,4]))]
```

Comment:
It doesn't seem to me like this is the correct expected case (and for the test below)? 
- eg. shouldn't the expected case (even though it'll always panic) be `U256([U64::MAX;4])`?

It's not obvious to me what's throwing the panic here - is it line 153 or 154? If it's 154, it would be better to not panic and `assert_eq` explicitly against the `U512toU256Error::Overflow` that's being returned for overflows.

Reply:
Well in my impl if there is an overflow the conversion just fails and doesn't default to the max u256 value, so I changed this test case to have an expected value of `Ut12toU256Error::Overflow` as that should be the actual expected value. Let me know if this is what you were thinking.